modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/MessageBoardMessageResourceImpl.java

```diff
@@ -344,19 +344,28 @@ public MessageBoardMessage putMessageBoardThreadMessageBoardMessage(
 		MBThread mbThread = _mbThreadLocalService.getMBThread(
 			messageBoardThreadId);
 
-		MBMessage mbMessage =
-			_mbMessageLocalService.fetchMBMessageByReferenceCode(
-				mbThread.getGroupId(), externalReferenceCode);
+		if (messageBoardMessage.getParentMessageBoardMessageId() == null) {
+			messageBoardMessage.setParentMessageBoardMessageId(
+				mbThread.getRootMessageId());
+		}
 
-		if (mbMessage == null) {
-			messageBoardMessage.setExternalReferenceCode(externalReferenceCode);
+		return _upsertMessageBoardMessageByExternalReferenceCode(
+			externalReferenceCode, mbThread.getGroupId(), messageBoardMessage);
+	}
 
-			return _addMessageBoardMessage(
-				mbThread.getRootMessageId(), messageBoardMessage);
+	@Override
+	public MessageBoardMessage putSiteMessageBoardMessage(
+			Long siteId, String externalReferenceCode,
+			MessageBoardMessage messageBoardMessage)
+		throws Exception {
+
+		if (messageBoardMessage.getParentMessageBoardMessageId() == null) {
```

Comment:
mmm... I thought this was the endpoint to create a message directly without a thread, is it a thread needed already?

Reply:
Oh, I see... we didn't have a site/messages PUT/POST because it always depends on a thread.

Reply:
Yes, my impression was that the thread was always needed, so I had to enforce this condition.

That is why at first I created the other endpoint, because I thought here was not possible without the thread id, but then I saw that parentMessageBoardMessageId was a property and was not read-only, so I implemented it this way